Libraries/Utilities/defineLazyObjectProperty.js

```diff
@@ -35,7 +35,21 @@ function defineLazyObjectProperty<T>(
     // `setValue` which calls `Object.defineProperty` which somehow triggers
     // `getValue` again. Adding `valueSet` breaks this loop.
     if (!valueSet) {
-      setValue(get());
+      try {
+        // Set `valueSet` to true optimistically to not fail into another infinite
+        // loop when property getter would try to access already defined property.
+        // This is specifically the case with 'regenerator-runtime' module, which
+        // checks if `global.regeneratorRuntime` property is already defined and
+        // thus triggers `getValue()` again.
+        valueSet = true;
```

Comment:
I don't think that this particular issue is caused by regenerator though. When I saw this myself, it was actually the line https://github.com/facebook/react-native/blob/master/Libraries/Core/InitializeCore.js#L156 which caused the recursion: The access of `global.regneratorRuntime` called into the same getter function.